src/classes/ALLO_AllocationsUtil.cls

```diff
@@ -0,0 +1,1939 @@
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Allocations
+* @group-content ../../ApexDocContent/Allocations.htm
+* @description Utility class to support shared functionality for Allocations
+*/
+public class ALLO_AllocationsUtil {
+    /** @description The number of times it will retry the set of Database operations, after first removing any with errors. */
+      private static final Integer NUM_DATABASE_RETRIES = 1;
+
+    /** @description Used as a way to identify parts of Allocation Sync that may or may not be allowed to
+      *execute depending on earlier execution statuses */
+    public enum PROCESS_DEF {
+        ALL,
+        OPP_AMT_UPD,
+        OPP_ALLOC_UPD,
+        PMT_ALLOC,
+        PMT_ALLOC_RECALC,
+        PMT_ALLOC_INITIALIZE,
+        PMT_ALLOC_SYNC_TO_OPP_ALLOC,
+        OPP_ALLOC_SYNC_FINALIZED,
+        OPP_ALLOC_SYNC_NONFINALIZED
+    }
+
+    /** @description Allocations settings. */
+    public static Allocations_Settings__c ALLOCATION_SETTINGS = UTIL_CustomSettingsFacade.getAllocationsSettings();
+
+    /** @description Caches call to UserInfo.IsMultiCurrencyOrganization */
+    private static final Boolean IS_MULTICURRENCY_ORG = UserInfo.isMultiCurrencyOrganization();
+
+    // Lazy initialization on use to reduce describes for Object Labels
+    /** @description Caches The Allocation Object's Label */
+    @testVisible private static String ALLOC_OBJ_LABEL {
+        get {
+            if (ALLOC_OBJ_LABEL == null) {
+                ALLOC_OBJ_LABEL = UTIL_Describe.getObjectLabel('Allocation__c');
```

Comment:
I believe it's necessary to apply the namespace to this. See the `ALLO_ManageAllocations_CTRL` class for an example.